src/flow_solver/flow_solver_cases/periodic_turbulence.cpp

```diff
@@ -84,6 +127,195 @@ double PeriodicTurbulence<dim,nstate>::get_adaptive_time_step(std::shared_ptr<DG
+    std::string mpi_rank_string = std::to_string(mpi_rank_input);
+    const unsigned int length_of_mpi_rank_with_padding = 5;
+    const int number_of_zeros = length_of_mpi_rank_with_padding - mpi_rank_string.length();
+    mpi_rank_string.insert(0, number_of_zeros, '0');
+
+    return mpi_rank_string;
+}
+
+template<int dim, int nstate>
+void PeriodicTurbulence<dim, nstate>::output_velocity_field(
+    std::shared_ptr<DGBase<dim,double>> dg,
+    const unsigned int output_file_index,
+    const double current_time) const
+{
+    this->pcout << "     ... Writting velocity field ... " << std::flush;
+
+    // NOTE: Same loop from read_values_from_file_and_project() in set_initial_condition.cpp
+    
+    // Get filename prefix based on output file index and the flow field quantity filename prefix
+    const std::string filename_prefix = flow_field_quantity_filename_prefix + std::string("-") + std::to_string(output_file_index);
+
+    // (1) Get filename based on MPI rank
+    //-------------------------------------------------------------
+    // -- Get padded mpi rank string
+    const std::string mpi_rank_string = get_padded_mpi_rank_string(this->mpi_rank);
+    // -- Assemble filename string
+    const std::string filename_without_extension = filename_prefix + std::string("-") + mpi_rank_string;
+    const std::string filename = output_flow_field_files_directory_name + std::string("/") + filename_without_extension + std::string(".dat");
+    //-------------------------------------------------------------
+
+    // (1.5) Write the exact output time for the file to the table 
+    //-------------------------------------------------------------
+    if(this->mpi_rank==0) {
+        const std::string filename_for_time_table = output_flow_field_files_directory_name + std::string("/") + std::string("exact_output_times_of_velocity_field_files.txt");
+        // Add values to data table
+        this->add_value_to_data_table(output_file_index,"output_file_index",this->exact_output_times_of_velocity_field_files_table);
+        this->add_value_to_data_table(current_time,"time",this->exact_output_times_of_velocity_field_files_table);
+        // Write to file
+        std::ofstream data_table_file(filename_for_time_table);
+        this->exact_output_times_of_velocity_field_files_table->write_text(data_table_file);
+    }
+    //-------------------------------------------------------------
+
+    // (2) Read file
+    //-------------------------------------------------------------
+    std::ofstream FILE (filename);
+    
+    // check that the file is open and write DOFs
+    if (!FILE.is_open()) {
+        this->pcout << "ERROR: Cannot open file " << filename << std::endl;
+        std::abort();
+    } else if(this->mpi_rank==0) {
+        const unsigned int number_of_degrees_of_freedom_per_state = dg->dof_handler.n_dofs()/nstate;
+        FILE << number_of_degrees_of_freedom_per_state << std::string("\n");
```

Comment:
Do you want these files to be readable by a human? You could add standard headers and modify the file reading function to match. Up to you though.

Reply:
we can leave it as is, I'd have to update all the files used for the tests and what I've been running so far

Reply:
For sure, it's a matter of preference.